docs/user-guide/scaling.rst

```diff
@@ -0,0 +1,319 @@
+scaling MongoDB is available in the `MongoDB Deployment`_ docs.
+
+Pulp uses either RabbitMQ or Apache Qpid as its messaging backend. Pulp does
+not generate many messages in comparison to other applications, so it is not
+expected that the messaging backend would need to be scaled for performance
+unless the number of concurrent consumer connections is large. However,
+additional configuration may be done to make the messaging backend more fault
+tolerant. Examples of this are available in the `Apache Qpid HA docs`_ and
+the `RabbitMQ HA docs`_.
+
+.. warning:: There is a bug in versions of Apache Qpid older than 0.30 that
+    involves running out of file descriptors. This is an issue on deployments
+    with large numbers of consumers. See
+    `RHBZ #1122987 <https://bugzilla.redhat.com/show_bug.cgi?id=1122987>`_
+    for more information about this and for suggested workarounds.
+
+
+Scaling httpd
+-------------
+Additional httpd servers can be added to Pulp to increase both throughput
+and redundancy.
+
+In situations when there are more incoming HTTP or HTTPS requests than a single
+server can respond to it may be time to add additional httpd servers. httpd
+servers both the Pulp API and content so increasing capacity could improve
+both API and content delivery performance.
+
+Consider using the Apache `mod_status`_ scoreboard to monitor how busy your
+httpd workers are.
+
+.. note::
+    Pulp itself does not provide httpd load balancing capabilities. See the
+    `Load Balancing Requirements`_ for more information.
+
+To add additional httpd server capacity, configure the desired number of
+`Pulp clustered servers` and start ``httpd`` on them. Remember only one
+instance of ``pulp_celerybeat`` and ``pulp_resource_manager`` should be
+running across all `Pulp clustered servers`.
+
+
+Scaling workers
+---------------
+
+Additional Pulp workers can be added to increase asynchronous work throughput
+and redundancy.
+
+To add additional Pulp worker capacity, configure the desired number of `Pulp
+clustered servers` according to the the `clustering`_ docs and start
+``pulp_workers`` on each of them. Remember only one instance of
+``pulp_celerybeat`` and ``pulp_resource_manager`` should be running across
+all `Pulp clustered servers`.
+
+
+.. _clustering:
+
+Clustering Pulp
+---------------
+
+A clustered Pulp installation is comprised of two or more `Pulp clustered
+servers`. The term `Pulp clustered server` is used to distinguish it as a
+separate concept from :ref:`pulp_nodes`. `Pulp clustered servers` share the
+following components:
+
++--------------------+---------------------------------------------------------+
+| Pulp Configuration | Pulp reads its configuration from conf files inside     |
+|                    | ``/etc/pulp``.                                          |
++--------------------+---------------------------------------------------------+
+| Pulp Content       | Pulp requires synced content available on all clustered |
```

Comment:
I'd make this more general, since we store other stuff in /var/lib/pulp besides just unit files. Perhaps call it "Pulp Files" and "Pulp stores files on disk within /var/lib/pulp". We don't want anyone going in there and mucking around, so I think it's fine to keep it vague.